doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,587 @@
+        + major (number) - The currently used major version
+        + minor (number) - The currently used minor version
+
+
+
+# Authentication endpoint [/auth]
+
+## Authenticate [POST]
+
+The API will try to authenticate the user by the submitted credentials. In case of success a session token, which can be used for further requests to the API, will be returned.
+
++ Request (application/json)
+    + Attributes (object)
+        + username: `the-username` (string, required)
+        
+            The name of the user account which should be tried to authenticate.
+
+            Regex: ^[a-zA-Z0-9\-\.]{3,20}$
+        
+        + password: `a5ecretP4ssword` (string, required)
+        
+            The corresponding password for the supplied user account.
+ 
++ Response 200 (application/json)
+    + Attributes (object)
+        + token (string) - The session token which can be used to authenticate against the server in further requests
+
++ Response 400 (application/json)
+    + Attributes (object)
+        + status (string) - Textual description of the HTTP status
+        + message (string) - Detailed error information, e.g. hint about malformed request
+        + i18n (string) - A unique token representing above error description message; can be used for internationalization in frontends
+
+            A comprehensive list of possible errors:
+            - `AUTH_MISSING_USERNAME` if no input for `username` was submitted
+            - `AUTH_MISSING_PASSWORD` if no input for `password` was submitted
+
++ Response 401 (application/json)
+    + Attributes (object)
+        + status (string) - Textual description of the HTTP status
+        + message (string) - Detailed error information, e.g. hint about malformed request
+        + i18n (string) - A unique token representing above error description message; can be used for internationalization in frontends
+
+            A comprehensive list of possible errors:
+            - `AUTH_UNKNOWN_USERNAME` if there is no user registered witht he given name
+            - `AUTH_INVALID_PASSWORD` if the given password does not match the password saved for the given account
+
+
+# Users endpoint [/users{?rows}{?sort}{?sortby}{?filter}{?filterby}{?current}]
+
+## User list [GET]
+
+Retrieves a list of users. The result can be adjusted by additional, but optional parameters. This operation requires moderator permissions (rank 2) or higher.
+
++ Parameters
+    + rows (number, optional) - How many entries will be added to the output at max.
+        + Default: `200`
+    + sort (enum[string], optional) - How to sort the entries for the output.
+        + Default: `asc`
+        + Members
+            + `asc`
+            + `desc`
+    + sortby (enum[string], optional) - What criteria should be used for sorting
+        + Default: `username`
+        + Members
+            + `username` - The users name
+            + `email` - email
+            + `created_at` - Creation date
```

Comment:
It is a timestamp in integer format, but here it is only an enum for the parameter, so no further specification necessary. I don't think knowing the registration date is essential, but having it is often useful for statistics and other stuff (e.g. to know who has been on board for a long time and could get some more privilegues...).